charts/iap/templates/deployments.yaml

```diff
@@ -83,7 +90,7 @@ spec:
       nodeSelector:
 {{ toYaml $.Values.iap.nodeSelector | indent 8 }}
       affinity:
-{{ (tpl (toYaml $.Values.iap.affinity) (merge $ .)) | fromYaml | toYaml | indent 8 }}
+{{ (tpl (toYaml $.Values.iap.affinity) (merge (deepCopy $) .)) | fromYaml | toYaml | indent 8 }}
```

Comment:
This fixes a problem whre settings from one Deployment would affect all other deployments, i.e. if Grafana sets `pass_user_headers`, suddenly Alertmanager and Prometheus have the same value.